src/lib/x509/certstor.cpp

```diff
@@ -100,6 +101,7 @@ std::optional<X509_Certificate> Certificate_Store_In_Memory::find_cert_by_pubkey
    }
 
    auto hash = HashFunction::create("SHA-1");
+   BOTAN_ASSERT_NONNULL(hash);
```

Comment:
IMO either this should be  `create_and_throw`, or else add a comment here to the effect of
> The sha1 module is a hard dependency of x509 so the hash should always be available
[Fine with either]